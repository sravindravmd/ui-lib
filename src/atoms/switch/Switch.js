import React, { useState, useCallback } from 'react'
import {
  StyledSwitchWrapper,
  StyledToggleSwitch,
  StyledCheckbox,
  Slider,
  Styledtext,
} from './Switch.styles'
import PropTypes from 'prop-types'

export const Switch = ({ initialState, label, ontoggle, ...props }) => {
  const [toggle, setToggle] = useState(initialState)

  const handleToggle = useCallback(
    e => {
      setToggle(e.target.checked)
      ontoggle(e.target.checked)
    },
    [ontoggle]
  )

  return (
    <StyledSwitchWrapper>
      <StyledToggleSwitch>
        <StyledCheckbox
          {...props}
          data-testid='toggle'
          type='checkbox'
          checked={toggle}
          value={toggle}
          onChange={e => handleToggle(e)}
        />
        <Slider toggle={toggle}></Slider>
      </StyledToggleSwitch>
      {label && <Styledtext>{label}</Styledtext>}
    </StyledSwitchWrapper>
  )
}

Switch.defaultProps = {
  label: '',
  ontoggle: () => {},
  initialState: false,
}

Switch.propTypes = {
  label: PropTypes.string,
  ontoggle: PropTypes.func,
  initialState: PropTypes.bool,
}
