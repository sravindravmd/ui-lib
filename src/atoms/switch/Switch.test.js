import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'
import { Switch } from './Switch'

afterEach(cleanup)

describe('<Switch/>', () => {
  it('renders Switch with initialState:false correctly', () => {
    const { container, getByText } = render(<Switch label='Label' />)
    expect(getByText('Label')).toHaveTextContent('Label')
    expect(container).toMatchSnapshot()
  })

  it('renders Switch with initialState:true correctly', () => {
    const { container, getByText } = render(
      <Switch initialState={true} label='Label' />
    )
    expect(getByText('Label')).toHaveTextContent('Label')
    expect(container).toMatchSnapshot()
  })

  it('renders Switch with initialState:true correctly', () => {
    const { getByTestId, getByText } = render(<Switch label='Label' />)
    fireEvent.click(getByText('Label'))
    expect(getByTestId('toggle')).toBeChecked()
    fireEvent.click(getByText('Label'))
    expect(getByTestId('toggle')).not.toBeChecked()
  })
})
