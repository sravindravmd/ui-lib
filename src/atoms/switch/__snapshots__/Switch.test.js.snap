// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch/> renders Switch with initialState:false correctly 1`] = `
<div>
  <label
    class="sc-bdfBwQ eQApsv"
  >
    <span
      class="sc-gsTCUz edZUkl"
    >
      <input
        class="sc-dlfnbm lcKuwO"
        data-testid="toggle"
        type="checkbox"
        value="false"
      />
      <span
        class="sc-hKgILt iYTDUb"
      />
    </span>
    <span
      class="sc-eCssSg bddYCL"
    >
      Label
    </span>
  </label>
</div>
`;

exports[`<Switch/> renders Switch with initialState:true correctly 1`] = `
<div>
  <label
    class="sc-bdfBwQ eQApsv"
  >
    <span
      class="sc-gsTCUz edZUkl"
    >
      <input
        checked=""
        class="sc-dlfnbm lcKuwO"
        data-testid="toggle"
        type="checkbox"
        value="true"
      />
      <span
        class="sc-hKgILt bnRFyi"
      />
    </span>
    <span
      class="sc-eCssSg bddYCL"
    >
      Label
    </span>
  </label>
</div>
`;
